### PR TITLE
auction house basic POC and finding nft mints

### PR DESCRIPTION
**Pull request by atharmohammad**
Implements :
1). POC for auction house
2). finding all minted nfts for a candymachine

``` Currently has some hardcoded keypairs for testing purposes ```

### CODE REVIEW

File: src/vendors/auction-house.service.ts [hunk 1]
```diff
@@ -18,13 +27,22 @@ export class AuctionHouseService {
...
+    //   Buffer.from(JSON.parse(treasuryWallet.toString(Utf8))),
+    // );
+    const treasuryKeypair: Keypair = Keypair.fromSecretKey(
+      new Uint8Array([
+        193, 175, 203, 110, 116, 69, 233, 189, 129, 146, 244, 26, 38, 246, 84,
+        86, 129, 192, 248, 25, 62, 249, 10, 3, 152, 68, 88, 16, 13, 27, 182, 10,
+        47, 249, 117, 244, 173, 148, 158, 132, 48, 71, 199, 138, 145, 178, 194,
+        132, 56, 56, 174, 35, 108, 239, 223, 54, 150, 232, 194, 12, 224, 56,
+        171, 92,
+      ]),
```
Comment (josip-volarevic) [inline src/vendors/auction-house.service.ts:45]: What not get the key from the `.env` file? Do you need help hashing the private key?

File: src/vendors/auction-house.service.ts [hunk 2]
```diff
@@ -53,14 +75,14 @@ export class AuctionHouseService {
     }
   }
 
-  async findOurAuctionHouse() {
-    const address = new PublicKey('something');
+  async findOurAuctionHouse(address: PublicKey) {
+    // const address = new PublicKey('something');
     return this.metaplex.auctionHouse().findByAddress({ address });
   }
```
Comment (josip-volarevic) [inline src/vendors/auction-house.service.ts:81]: I was thinking we could add the `private readonly auctionHouseAuthority: Keypair;`

In that case, we could have this function work as
```ts
  async findOurAuctionHouse() {
    return this.metaplex.auctionHouse().findByAddress({ address: this.auctionHouseAuthority.publicKey });
  }
```

Or if we accept the `publicKey` parameter
```ts
  async findAuctionHouseByAddress(publicKey: PublicKey) {
    return this.metaplex.auctionHouse().findByAddress({ address });
  }
```

File: src/vendors/auction-house.service.ts (hunk 1 above)
Comment (atharmohammad) [reply src/vendors/auction-house.service.ts:45]: yeah! not sure was getting parse error and i just thought to solve it at the end and move on with important parts first.will resolve it in the call

**Comment by josip-volarevic** [comment]
Closing this PR as progress was moved to #7 
Credits to @atharmohammad

**Comment by josip-volarevic** [comment]
Closing this PR as progress was moved to #7 
Credits to @atharmohammad

**Pull request closed without merging**